File: custom_dialog/complexCustDialog/app/src/main/java/com/cxyzy/customdialog/MainActivity.java
package com.cxyzy.customdialog;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.button1).setOnClickListener(v -> {
            new VoiceDetectDialog(MainActivity.this, VoiceDetectDialog.Styles.LISTENING).show();
        });
        findViewById(R.id.button2).setOnClickListener(v -> {
            new VoiceDetectDialog(MainActivity.this, VoiceDetectDialog.Styles.DETECTING).show();
        });
        findViewById(R.id.button3).setOnClickListener(v -> {
            new VoiceDetectDialog(MainActivity.this, VoiceDetectDialog.Styles.RESULT_FAILED).show();
        });
    }
}

File: custom_dialog/simpleBottomDialog/app/src/main/java/com/cxyzy/simplebottomdialog/MainActivity.kt
package com.cxyzy.simplebottomdialog

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        button1.setOnClickListener { BottomDialog(this).show() }
        button2.setOnClickListener { BottomButtonDialog(this).show() }
        button3.setOnClickListener { BottomDialogWithAnim(this).show() }
        button4.setOnClickListener { BottomDialogWithAnim1(this).show() }
    }
}
